### ISSUE TITLE
Invalid references with .NET Framework v4.6.1+

### ISSUE THREAD

**Issue by iainnicol** [fsprojects/Paket#2764]
### Description

I end up with invalid references when I target .NET Framework v4.6.1, .NET Framework v4.6.2, or .NET Framework v4.7.  In contrast, targetting .NET Framework v4.6 usually (always?) seems to work. 

### Repro steps

My Paket is up-to-date, at version 5.96.1.

I have Visual Studio 15.3.4 is installed.  So I think the .NET Standard 2.0 facades for .NET Framework v4.6.1 are installed.

Create a C# or F# project targetting v4.6.1 or higher.

paket.dependencies:
```
source https://nuget.org/api/v2

nuget Microsoft.CodeAnalysis.Common = 2.3.2
```

paket.references:
```
Microsoft.CodeAnalysis.Common
```

Run:
``./.paket/paket.exe auto-restore on; ./.paket/paket.exe install``

### Expected behavior

The references created are valid.

### Actual behavior

Some of the references created are bad.  Specifically, in Visual Studio, the references appear with a yellow warning icon, and the error list says "The referenced component 'foo' could not be found".

The following references are bad:
```
System.AppContext
System.Console
System.Diagnostics.FileVersionInfo
System.Diagnostics.StackTrace
System.IO
System.IO.Compression
System.IO.FileSystem
System.IO.FileSystem.Primitives
System.Linq
System.Linq.Expressions
System.Reflection
System.Runtime
```
and several others.

### Known workarounds

Raw NuGet doesn't have this problem.

**Comment by forki** [comment]
Can you please upload a zip with that sample project?

Am 14.09.2017 12:22 schrieb "Iain Nicol" <notifications@github.com>:

**Comment by iainnicol** [comment]
No problem.  Well, I put a repo up at https://github.com/iainnicol/paket-net461-bug

**Comment by forki** [comment]
something look completely strange here. From first look into the fsproj the generated nodes look ok, but VS is confused for some reason. /cc @matthid any ideas?

**Comment by Bettzu** [comment]
We have the same problem when we switch to .net 4.6.1. However, only in conjunction with dependency groups in the package.dependencies and .refence files. Are there any further facts about this topic?

**Comment by matthid** [comment]
More information about why 4.6.1 is special can be found [here](https://github.com/fsprojects/Paket/pull/2391)

Maybe NuGet has some special magic to remove framework packages from the dependencies list. If you look at https://www.nuget.org/packages/Microsoft.CodeAnalysis.Common/2.3.2# you can clearly see how the package has all the dependencies you list as "bad".

Not sure why they are "bad", maybe a Visual Studio bug?

**Comment by 0x53A** [comment]
1) If you can, try to target net471. This is the first version with "native" support for netstandard2.0.

2) These packages overriding framework-included references should be ok. They have a higher version, so the runtime will load them instead of the in-box ones.
The downside is that you need to deploy these with your app, and that you need to make sure your app.config has the correct binding redirects.

3) your repro works for me without issues. No idea what the difference is, I have the latest version of VS and a relatively recent version of dotnet-cli. Make sure to first restore packages, THEN open VS, just as a sanity check.

**Comment by iainnicol** [comment]
@0x53A wrote:
> your repro works for me without issues. No idea what the difference is, I have the latest version of VS and a relatively recent version of dotnet-cli. Make sure to first restore packages, THEN open VS, just as a sanity check.

Weirdly, I still get the issue in Visual Studio 15.4.5, if targetting net461.  Even after following your ordering, I still see the yellow/invalid references.

> If you can, try to target net471. This is the first version with "native" support for netstandard2.0.

But net471 (.NET Framework v4.7.1) fixes the issue for me. Yay.

@Bettzu, hopefully net471 fixes your issues as well.  I'm optimistically closing on that assumption.

**Comment by 0x53A** [comment]
> Weirdly, I still get the issue in Visual Studio 15.4.5, if targetting net461. Even after following your ordering, I still see the yellow/invalid references.

![grafik](https://user-images.githubusercontent.com/4236651/33552293-c407b2c0-d8f4-11e7-875e-d1c91cefb3e4.png)

**Comment by iainnicol** [comment]
@0x53A 
> I should probably have specified that I tested it in the latest (at that time) preview. ;-)

Ah.  That explains it.  So the good news is we are now both seeing consistent behaviour.  And more importantly, this bug is fixed in Visual Studio 15.5.0 Preview 4.
